Core/NoOpRunner.Core/PlayerStates/PlayerOneStateMachine.cs

```diff
@@ -1,29 +1,50 @@
 ﻿using System;
 using NoOpRunner.Core.Enums;
+using NoOpRunner.Core.Shapes;
 
 namespace NoOpRunner.Core.PlayerStates
 {
     public class PlayerOneStateMachine
     {
-        private PlayerOneState currentState = PlayerOneState.Idle;
-        
+        private PlayerState CurrentState;
+
+        private PlayerState LastState { get; set; }
+
         public PlayerOneState State
```

Comment:
It would be nice if you tested your code before submitting it for a review. Host-client only syncs the `State` right now, so host crashes on initial connection. But even besides that, keeping both the state enum and the state looks smelly. Keeping only the state should suffice. Refactor, make sure the `PlayerState` is synced properly, remove the `getEnum()` from `PlayerState`

Reply:
Fixed